<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
 <HEAD>
   <TITLE> [Mono-bugs] [Bug 590488] New: Not an exception when connecting to	the socket.
   </TITLE>
   <LINK REL="Index" HREF="index.html" >
   <LINK REL="made" HREF="mailto:mono-bugs%40lists.ximian.com?Subject=%5BMono-bugs%5D%20%5BBug%20590488%5D%20New%3A%20Not%20an%20exception%20when%20connecting%20to%0A%09the%20socket.&In-Reply-To=">
   <META NAME="robots" CONTENT="index,nofollow">
   <META http-equiv="Content-Type" content="text/html; charset=us-ascii">
   <LINK REL="Previous"  HREF="098627.html">
   <LINK REL="Next"  HREF="098467.html">
 </HEAD>
 <BODY BGCOLOR="#ffffff">
   <H1>[Mono-bugs] [Bug 590488] New: Not an exception when connecting to	the socket.</H1>
    <B>bugzilla_noreply at novell.com</B> 
    <A HREF="mailto:mono-bugs%40lists.ximian.com?Subject=%5BMono-bugs%5D%20%5BBug%20590488%5D%20New%3A%20Not%20an%20exception%20when%20connecting%20to%0A%09the%20socket.&In-Reply-To="
       TITLE="[Mono-bugs] [Bug 590488] New: Not an exception when connecting to	the socket.">bugzilla_noreply at novell.com
       </A><BR>
    <I>Tue Mar 23 09:54:26 EDT 2010</I>
    <P><UL>
        <LI>Previous message: <A HREF="098627.html">[Mono-bugs] [Bug 590436] mmap failure in System.GC.Internal.Collect() on Xen - x86-64
</A></li>
        <LI>Next message: <A HREF="098467.html">[Mono-bugs] [Bug 590488] Not an exception when connecting to the	socket.
</A></li>
         <LI> <B>Messages sorted by:</B> 
              <a href="date.html#98458">[ date ]</a>
              <a href="thread.html#98458">[ thread ]</a>
              <a href="subject.html#98458">[ subject ]</a>
              <a href="author.html#98458">[ author ]</a>
         </LI>
       </UL>
    <HR>  
<!--beginarticle-->
<PRE><A HREF="http://bugzilla.novell.com/show_bug.cgi?id=590488">http://bugzilla.novell.com/show_bug.cgi?id=590488</A>

<A HREF="http://bugzilla.novell.com/show_bug.cgi?id=590488#c0">http://bugzilla.novell.com/show_bug.cgi?id=590488#c0</A>


           Summary: Not an exception when connecting to the socket.
    Classification: Mono
           Product: Mono: Class Libraries
           Version: 2.6.x
          Platform: x86
        OS/Version: Windows XP
            Status: NEW
          Severity: Normal
          Priority: P5 - None
         Component: System
        AssignedTo: <A HREF="http://lists.ximian.com/mailman/listinfo/mono-bugs">mono-bugs at lists.ximian.com</A>
        ReportedBy: <A HREF="http://lists.ximian.com/mailman/listinfo/mono-bugs">s.lipatov at gmail.com</A>
         QAContact: <A HREF="http://lists.ximian.com/mailman/listinfo/mono-bugs">mono-bugs at lists.ximian.com</A>
          Found By: ---
           Blocker: No


Created an attachment (id=350043)
 --&gt; (<A HREF="http://bugzilla.novell.com/attachment.cgi?id=350043">http://bugzilla.novell.com/attachment.cgi?id=350043</A>)
Execution of code by native .NET platform and by Mono.

Description of Problem:
Have a code:
When server is not started (not listening port 5555) no exception is generated.

&gt;<i> &gt; &gt; using System;
</I>&gt;<i> &gt; &gt; using System.Net.Sockets;
</I>&gt;<i> &gt; &gt; namespace ConsoleApplication1
</I>&gt;<i> &gt; &gt; {
</I>&gt;<i> &gt; &gt;     class Program
</I>&gt;<i> &gt; &gt;     {
</I>&gt;<i> &gt; &gt;         static void Main()
</I>&gt;<i> &gt; &gt;         {
</I>&gt;<i> &gt; &gt;             Console.WriteLine(&quot;Hello, threads!&quot;);
</I>&gt;<i> &gt; &gt;             try
</I>&gt;<i> &gt; &gt;             {
</I>&gt;<i> &gt; &gt;                 (new Socket(AddressFamily.InterNetwork,
</I>&gt;<i> &gt; SocketType.Stream,
</I>&gt;<i> &gt; &gt; ProtocolType.Tcp)).Connect(&quot;127.0.0.1&quot;, 5555);
</I>&gt;<i> &gt; &gt;             }
</I>&gt;<i> &gt; &gt;             catch (Exception e)
</I>&gt;<i> &gt; &gt;             {
</I>&gt;<i> &gt; &gt;                 System.Console.WriteLine(e.Message);
</I>&gt;<i> &gt; &gt;             }
</I>&gt;<i> &gt; &gt;             Console.ReadKey();
</I>&gt;<i> &gt; &gt;         }
</I>&gt;<i> &gt; &gt;     }
</I>&gt;<i> &gt; &gt; }
</I>

Steps to reproduce the problem:
1. Compile
2. Run


Actual Results:
Hello, threads!

Expected Results:
Hello, threads!
Exception message: Connection is not established...

How often does this happen? 
Always

Additional Information: running configuration - Mono 2.6.3 for Windows, Windows
XP SP3. screenshot is attached.

-- 
Configure bugmail: <A HREF="http://bugzilla.novell.com/userprefs.cgi?tab=email">http://bugzilla.novell.com/userprefs.cgi?tab=email</A>
------- You are receiving this mail because: -------
You are the QA contact for the bug.
You are the assignee for the bug.
</PRE>














































<!--endarticle-->
    <HR>
    <P><UL>
        <!--threads-->
	<LI>Previous message: <A HREF="098627.html">[Mono-bugs] [Bug 590436] mmap failure in System.GC.Internal.Collect() on Xen - x86-64
</A></li>
	<LI>Next message: <A HREF="098467.html">[Mono-bugs] [Bug 590488] Not an exception when connecting to the	socket.
</A></li>
         <LI> <B>Messages sorted by:</B> 
              <a href="date.html#98458">[ date ]</a>
              <a href="thread.html#98458">[ thread ]</a>
              <a href="subject.html#98458">[ subject ]</a>
              <a href="author.html#98458">[ author ]</a>
         </LI>
       </UL>

<hr>
<a href="http://lists.ximian.com/mailman/listinfo/mono-bugs">More information about the mono-bugs
mailing list</a><br>
</body></html>
